pit30m/data/log_reader.py

```diff
@@ -143,24 +148,50 @@ def __repr__(self) -> str:
         return f"Pit30M Log Reader: {self._log_root_uri}"
 
     @cached_property
-    def partitions_mask(self) -> np.ndarray:
+    def _partitions_mask(self) -> np.ndarray:
         """
         Returns a boolean np array that accounts for the requested partitions, by setting sensor readings that should
-        be skipped to False. Currently computed from the Front Camera.
+        be skipped to False. Computed for the raw pose data @ 100 Hz.
+
+        Returns:
+            A structured array with two fields:
+            1. "capture_time": float 64 series of pose GPS timestamps
+            2. "mask": boolean indicating whether each pose is included in the requested partitions
         """
-        if not self.partitions:
-            n_sensor_measurements = len(self.get_cam_geo_index(CamName.MIDDLE_FRONT_WIDE))
-            return np.full(n_sensor_measurements, True)
+        # breakpoint()
+        n_sensor_measurements = len(self.raw_pose_data)
+        partition_indices = self._partition_assigments
+        partition_indices += (np.full(n_sensor_measurements, True),)
+
+        combined_indices_mask = np.logical_and.reduce(partition_indices)
+        timestamps = self.raw_pose_data["capture_time"]
+
+        # HACK: Convert the raw poses' timestamps to UNIX to make them comparable to those... everywhere else
```

Comment:
Could you guard this so we only do it for v1? Indexes v2 and on will use GPS time as discussed ofline.

Reply:
Would you be ok with re-naming the indices to v0 before the dataset is announced? I'd rather not have custom logic for versions that people are not supposed to use.

Reply:
Moved to v2.